hazelcast/src/main/java/com/hazelcast/core/Offloadable.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.core;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.ExecutorConfig;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.spi.ExecutionService;
+
+/**
+ * Allows off-loading the processing unit implementing this interface to the specified or default Executor.
+ *
+ * Currently supported in:
+ * <ul>
+ *     <li>{@link IMap#executeOnKey(Object, EntryProcessor)}</li>
+ *     <li>{@link IMap#submitToKey(Object, EntryProcessor)} </li>
+ *     <li>{@link IMap#submitToKey(Object, EntryProcessor, ExecutionCallback)} </li>
+ * </ul>
+ */
+public interface Offloadable {
+
+    /**
+     * Constant meaning that there will be no off-loading if returned from the {@link #getExecutorName()} method.
+     */
+    String NO_OFFLOADING = "a2b7e85f-17ce-4fff-8833-4516efeb0b74";
```

Comment:
Enes had this pretty sensible comment that it is possible to send a null or an empty string by coincidence, just forgetting to initialize that.